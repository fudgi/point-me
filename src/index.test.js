import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import App from './components/app_container';
import Sidebar from './components/sidebar/sidebar';
import SidebarElement from './components/sidebar_element/sidebar_element';

Enzyme.configure({ adapter: new Adapter() });

describe('Render Test', () => {
  it('App render check', () => {
    const output = Enzyme.shallow( <App />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('Sidebar render check', () => {
    const output = Enzyme.shallow( <Sidebar points={[]} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('SidebarElement render check', () => {
    const output = Enzyme.shallow( <SidebarElement point={{coordinates:[10,30], name: "", color: ""}} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('Sidebar Click Test', () => {
  const mockCallBack = jest.fn();
  const output = Enzyme.mount( <Sidebar points={[]} addNewPoint={mockCallBack} resetAllPoints={mockCallBack}/>);
    it('point add check', () => {
      output.find('.sidebar-add').simulate('click', {preventDefault: () => {}})
      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack.mock.calls[0][0]).toBe("");
    });
    it('reset points check', () => {
      output.find('.sidebar_reset').simulate('click', () => {
        expect(mockCallBack.mock.calls.length).toEqual(1);
      })
    });
});

describe('SidebarElement Event Test', () => {
  const mockCallBack = jest.fn();
  const output = Enzyme.mount( <SidebarElement point={{coordinates:[10,30], name: "", color: ""}} deletePoint={mockCallBack} onDragStart={mockCallBack}/> );
  it('delete click check', () => {
    output.find('.sidebar_element_button').simulate('click', () => {
      console.log('dasdw')
      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack.mock.calls[0][0]).toBe("");
    })
  });
  it('drugstart test', () => {
    output.find('.sidebar_element').simulate('dragstart', () => {
      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack.mock.calls[0][1]).toBe(Number);
    })
  });
});

describe('App test', () => {

  const output = Enzyme.mount( <App /> );
  const instance = output.instance();

  it('add point out of state', () => {
    instance.addNewPoint("name");
    expect(output.state('points').length).toEqual(1);
  })

  it('update point', () =>{
    instance.updatePointCoords([10,30], 0);
    expect(output.state('points')[0].coordinates).toEqual([10,30]);
  })

  it('delete point', ()=>{
    instance.deletePoint(0);
    expect(output.state('points').length).toEqual(0);
  })
  
  it('delete all points', () => {
    for(let i=0; i < 10; i++){
      instance.addNewPoint('');
    }
    expect(output.state('points').length).toEqual(10);
    instance.resetAllPoints();
    expect(output.state('points').length).toEqual(0);
  })
})
