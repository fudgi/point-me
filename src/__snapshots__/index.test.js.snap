// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test App render check 1`] = `
<Fragment>
  <div
    className="container"
  >
    <header
      className="app-header"
    >
      <div
        className="brand"
      >
        <img
          alt="brand logo"
          className="brand-logo"
          src="direction.svg"
        />
        <p
          className="brand-text"
        >
          Point-Me
        </p>
      </div>
      <img
        alt="sidebar.svg"
        className="sidebar-opener"
        onClick={[Function]}
        src="sidebar.svg"
      />
    </header>
    <main>
      <Sidebar
        addNewPoint={[Function]}
        deletePoint={[Function]}
        onDragEnd={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        points={Array []}
        resetAllPoints={[Function]}
        sidebarOpened={false}
      />
      <section
        className="map"
      >
        <div
          className="map-loading"
        >
          <img
            alt="loading.gif"
            src="loading.gif"
          />
          <p>
            Пожалуйста, подожите. Карта скоро появится
          </p>
        </div>
        <YandexMap
          points={Array []}
          updatePointCoords={[Function]}
        />
      </section>
    </main>
  </div>
</Fragment>
`;

exports[`Render Test Sidebar render check 1`] = `
<Fragment>
  <div
    className="background-container"
  />
  <aside
    className="sidebar sidebar_hidden"
  >
    <div
      className="sidebar_container"
    >
      <div
        className="sidebar-controls"
      >
        <form>
          <label
            htmlFor="point-name"
          >
            Введите имя точки: 
          </label>
          <input
            className="input sidebar_namefield"
            id="point-name"
            placeholder="Название точки"
            type="text"
          />
          <button
            className="button sidebar-add"
            onClick={[Function]}
          >
            Добавить новую точку в центр карты
          </button>
        </form>
      </div>
      <section
        className="sidebar_element_container"
      />
      <button
        className="button sidebar_reset"
        onClick={[Function]}
      >
        Сбросить все точки
      </button>
    </div>
  </aside>
</Fragment>
`;

exports[`Render Test SidebarElement render check 1`] = `
<div
  className="sidebar_element"
  draggable={true}
  onDragEnd={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
>
  <div
    className="sidebar_element_left"
  >
    <div
      className="sidebar_element_circle"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    >
       
    </div>
  </div>
  <div
    className="sidebar_element_right"
  >
    <div
      className="sidebar_element_row"
    >
      <p>
         
        NaN
        . 
        Точка
         
      </p>
      <button
        className="sidebar_element_button"
        onClick={[Function]}
      >
        <img
          alt="delete element"
          className="sidebar_element_img"
          src="trash.svg"
        />
      </button>
    </div>
    <p
      className="sidebar_element_coord"
    >
       
      10.000 : 30.000
       
    </p>
  </div>
</div>
`;
